db/history_trimming_iterator.h

```diff
@@ -25,7 +27,7 @@ class HistoryTrimmingIterator : public InternalIterator {
       return true;
     }
     Slice current_ts = ExtractTimestampFromKey(key(), cmp_->timestamp_size());
-    return cmp_->CompareTimestamp(current_ts, Slice(filter_ts_)) < 0;
+    return cmp_->CompareTimestamp(current_ts, Slice(filter_ts_)) <= 0;
```

Comment:
By the API's definition, data with ts <= filter_ts_ should be reserved, I added a unittest for the `=` case in the last commit.